### PR TITLE
Migrate to Shouldly

### LINKED ISSUE #149: Migrate towards one assertions framework
We should migrate to one assertion framework in unit tests.

The best choice atm appears to be Shouldly due to some missing functionality in FluentAssertions.

### PR DESCRIPTION
**Pull request by sungam3r**
fixes #149

@ArieGato I will retarget PR after merging #144.

### CODE REVIEW

**Comment by sungam3r** [comment]
I merged changes from `master`. I would like to note that I'm still do not see workflow running. It confuses and prevents from fixing bugs with CI.

**Comment by ArieGato** [comment]
@sungam3r what can I do to make the ci build fire for you?

**Comment by sungam3r** [comment]
I think this is because the rule requiring review. Just try to disable it.

**Comment by ArieGato** [comment]
I think the ci build is missing: on pull_request

**Comment by sungam3r** [comment]
Right. I have not reviewed workflow files yet.

File: src/Serilog.Sinks.RabbitMQ/Sinks/RabbitMQ/RabbitMQClientConfiguration.cs [hunk 1]
```diff
@@ -105,7 +105,7 @@ public RabbitMQClientConfiguration From(RabbitMQClientConfiguration config)
         SslOption = config.SslOption;
         AutoCreateExchange = config.AutoCreateExchange;
         MaxChannels = config.MaxChannels;
-        Hostnames = config.Hostnames.ToArray();
+        Hostnames = config.Hostnames.ToList();
```
Comment (sungam3r) [inline src/Serilog.Sinks.RabbitMQ/Sinks/RabbitMQ/RabbitMQClientConfiguration.cs:108]: @ArieGato HostNames is list on line 27.

File: Directory.Build.props [hunk 2]
```diff
@@ -2,7 +2,14 @@
...
   <PropertyGroup>
     <LangVersion>latest</LangVersion>
+    <GenerateAssemblyInfo>true</GenerateAssemblyInfo>
     <ImplicitUsings>enable</ImplicitUsings>
+    <RepositoryType>git</RepositoryType>
+    <DebugType>embedded</DebugType>
+    <PublishRepositoryUrl>true</PublishRepositoryUrl>
+    <EmbedUntrackedSources>true</EmbedUntrackedSources>
+    <!-- https://help.github.com/en/actions/configuring-and-managing-workflows/using-environment-variables -->
+    <ContinuousIntegrationBuild Condition="'$(GITHUB_ACTIONS)' == 'true'">True</ContinuousIntegrationBuild>
```
Comment (sungam3r) [inline Directory.Build.props:12]: I suggest to move potentially shared configuration into top-level Directory.Build.props file.

File: .github/workflows/ci-build.yml [hunk 3]
```diff
@@ -1,13 +1,16 @@
...
+    branches:
+      - master
   push:
     branches:
       - '**'        # matches every branch
       - '!master'   # excludes master
 
 env:
-  Configuration: Release
+  Configuration: Debug
```
Comment (ArieGato) [inline .github/workflows/ci-build.yml:13]: The environment should be Release

**Review by ArieGato — CHANGES_REQUESTED** [review]

File: .github/workflows/ci-build.yml (hunk 3 above)
Comment (sungam3r) [inline .github/workflows/ci-build.yml:13]:
```suggestion
  Configuration: Release
```

Comment (sungam3r) [reply .github/workflows/ci-build.yml]: done

**Review by ArieGato — APPROVED** [review]

**Pull request merged**